setup.py

```diff
@@ -181,6 +181,30 @@
 ]
 
 
+# Generate extras requires
+def update_reqires(filename, extras_require, recommended):
+    with open(filename) as requirements:
+        for line in requirements:
+            line = line.strip()
+            # Skip comments, inclusion or blank lines
+            if not line or line.startswith("-r") or line.startswith("#"):
+                continue
+            dependency, section = line.split("#")
+            dependency = dependency.strip()
+            section = section.strip()
+            if section not in extras_require:
+                extras_require[section] = [dependency]
+            else:
+                extras_require[section].append(dependency)
+            extras_require["all"].append(dependency)
+            if recommended:
+                extras_require["recommended"].append(dependency)
+
+
+extras_require = {"recommended": [], "all": []}
+update_reqires("requirements/optional.txt", extras_require, False)
```

Comment:
reqires?

Reply:
It's called `extras_require` in the setup args, so I kept it same here (see below in the patch).

Reply:
`reqires` != `requires`

Reply:
Sorry, I didn't notice that.